src/main/java/no/nav/arxaas/hierarchy/DateBasedHierarchyBuilder.java

```diff
@@ -0,0 +1,106 @@
+import com.fasterxml.jackson.annotation.JsonGetter;
+import com.google.common.collect.Iterables;
+import org.deidentifier.arx.DataType;
+import org.deidentifier.arx.aggregates.HierarchyBuilderDate;
+
+import javax.validation.constraints.NotNull;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.List;
+import java.util.Objects;
+import java.util.stream.Collectors;
+
+public class DateBasedHierarchyBuilder implements HierarchyBuilder {
+
+    /**
+     * Granularity from ARX
+     * Copied from ARX to retain modularity
+     */
+    public enum Granularity {
+
+        /**  Granularity */
+        SECOND_MINUTE_HOUR_DAY_MONTH_YEAR(HierarchyBuilderDate.Granularity.SECOND_MINUTE_HOUR_DAY_MONTH_YEAR),
+        /**  Granularity */
+        MINUTE_HOUR_DAY_MONTH_YEAR(HierarchyBuilderDate.Granularity.MINUTE_HOUR_DAY_MONTH_YEAR),
+        /**  Granularity */
+        HOUR_DAY_MONTH_YEAR(HierarchyBuilderDate.Granularity.HOUR_DAY_MONTH_YEAR),
+        /**  Granularity */
+        DAY_MONTH_YEAR(HierarchyBuilderDate.Granularity.DAY_MONTH_YEAR),
+        /**  Granularity */
+        WEEK_MONTH_YEAR(HierarchyBuilderDate.Granularity.WEEK_MONTH_YEAR),
+        /**  Granularity */
+        WEEK_YEAR(HierarchyBuilderDate.Granularity.WEEK_YEAR),
+        /**  Granularity */
+        MONTH_YEAR(HierarchyBuilderDate.Granularity.MONTH_YEAR),
+        /**  Granularity */
+        WEEKDAY(HierarchyBuilderDate.Granularity.WEEKDAY),
+        /**  Granularity */
+        WEEK(HierarchyBuilderDate.Granularity.WEEK),
+        /**  Granularity */
+        QUARTER(HierarchyBuilderDate.Granularity.QUARTER),
+        /**  Granularity */
+        YEAR(HierarchyBuilderDate.Granularity.YEAR),
+        /**  Granularity */
+        DECADE(HierarchyBuilderDate.Granularity.DECADE),
+        /**  Granularity */
+        CENTURY(HierarchyBuilderDate.Granularity.CENTURY),
+        /**  Granularity */
+        MILLENIUM(HierarchyBuilderDate.Granularity.MILLENIUM);
+
+        /** Format string */
+        private HierarchyBuilderDate.Granularity arxGranularity;
+
+        /**
+         * Creates a new instance
+         * @param arxGranularity ARX version of the Enum
+         */
+        Granularity(HierarchyBuilderDate.Granularity arxGranularity) {
+            this.arxGranularity = arxGranularity;
+        }
+
+        private HierarchyBuilderDate.Granularity arxGranularity(){
+            return arxGranularity;
+        }
+    }
+    @NotNull
+    private final String dateFormat;
+    @NotNull
+    private final List<Granularity> granularities;
+
+    @JsonCreator
+    public DateBasedHierarchyBuilder(String dateFormat, List<Granularity> granularities) {
+        Objects.requireNonNull(dateFormat, "dateFormat should not be null");
+        Objects.requireNonNull(granularities, "granularities should not be null");
+        if(Iterables.isEmpty(granularities)) throw new IllegalArgumentException("granularities cannot be empty");
```

Comment:
Siden dere allerede har sjekket granularities for null, så kan dere ditche google/guava for dette, og bare skrive
`if (granularities.isEmpty()) { throw .. }`